libkbfs/mdserver_remote.go

```diff
@@ -387,6 +388,20 @@ func (md *MDServerRemote) ShouldRetryOnConnect(err error) bool {
 	return !inputCanceled
 }
 
+// CheckReachability implements the MDServer interface.
+func (md *MDServerRemote) CheckReachability(ctx context.Context) {
```

Comment:
I've found that using `DialTimeout` has been a better way to determine the connection is down after getting one of these "reachability" notifications than running a ping on the current connection. As an example, if you pull the network cable, any additional pings on an existing connection will just hang (at least on macOS). Whereas, if we try to `Dial` we hit an error immediately and kill the connection. 

We want to kill the connection because there is another annoying problem on macOS where if you pull the network cable, and then plug it back in, you can get a weird error about trying to bind to the local port that is already in use. The best way I have found to counter this problem on the CORE side is to just kill the connection and try to reconnect (in the presence of this reachability notification). Here is an example of what that error looks like:

2017-04-18T10:07:20.667529-04:00 ▶ [DEBU kbfs mdserver_remote.go:291] 1682 MDServerRemote: ping error write tcp 192.168.1.7:60118->52.73.199.240
:443: write: can't assign requested address